### DISCUSSION TITLE
Run Error : SaltProc with OpenMC 0.13.3

### DISCUSSION THREAD

**Discussion by tsuiyong** [arfc/saltproc#200] [General]
Hello, @yardasol. I have installed SaltProc-0.5.dev on Ubuntu 20.04 according to the installation guide for a few days, and PyNE was not installed according to the environment.yml, OpenMC 0.13.3 was installed from the source code, and the Openmc Python library was installed in the saltproc-env environment. Although my serpent **version is 2.1.0**, I think saltproc-OpenMC coupled system will run successfully and generate some useful results. 

However, after I run the following commands :`python saltproc -i examples/msbr/msbr_openmc.json`, OpenMC was invoked and began to read XS, but I encountered an error: `ERROR: Could not find the nuclide U234 specified in tally 5 in any material`. 

The total running log is as follows:

(saltproc-env) cuiyong@Super-Server:~/Program_files/saltproc$ python saltproc -i examples/msbr/msbr_openmc.json
Initiating Saltproc:
	Restart = False
	Template File Path  = {'materials': '/home/cuiyong/Program_files/saltproc/examples/msbr/materials.xml', 'settings': '/home/cuiyong/Program_files/saltproc/examples/msbr/settings.xml'}
	Output HDF5 database Path = /home/cuiyong/Program_files/saltproc/examples/msbr/openmc_msbr_test/saltproc_results.h5

Step #1 has been started
/home/cuiyong/mambaforge3/envs/saltproc-env/lib/python3.12/site-packages/openmc/mixin.py:70: IDWarning: Another Filter instance already exists with id=1.
  warn(msg, IDWarning)
/home/cuiyong/mambaforge3/envs/saltproc-env/lib/python3.12/site-packages/openmc/mixin.py:70: IDWarning: Another Filter instance already exists with id=2.
  warn(msg, IDWarning)
Running openmc
Invalid MIT-MAGIC-COOKIE-1 keyInvalid MIT-MAGIC-COOKIE-1 key
                                %%%%%%%%%%%%%%%
                           %%%%%%%%%%%%%%%%%%%%%%%%
                        %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
                      %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
                    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
                   %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
                                    %%%%%%%%%%%%%%%%%%%%%%%%
                                     %%%%%%%%%%%%%%%%%%%%%%%%
                 ###############      %%%%%%%%%%%%%%%%%%%%%%%%
                ##################     %%%%%%%%%%%%%%%%%%%%%%%
                ###################     %%%%%%%%%%%%%%%%%%%%%%%
                ####################     %%%%%%%%%%%%%%%%%%%%%%
                #####################     %%%%%%%%%%%%%%%%%%%%%
                ######################     %%%%%%%%%%%%%%%%%%%%
                #######################     %%%%%%%%%%%%%%%%%%
                 #######################     %%%%%%%%%%%%%%%%%
                 ######################     %%%%%%%%%%%%%%%%%
                  ####################     %%%%%%%%%%%%%%%%%
                    #################     %%%%%%%%%%%%%%%%%
                     ###############     %%%%%%%%%%%%%%%%
                       ############     %%%%%%%%%%%%%%%
                          ########     %%%%%%%%%%%%%%
                                      %%%%%%%%%%%

                 | The OpenMC Monte Carlo Code
       Copyright | 2011-2023 MIT, UChicago Argonne LLC, and contributors
         License | https://docs.openmc.org/en/latest/license.html
         Version | 0.13.3
        Git SHA1 | 27cb0dc97960fe6d750eb5a93584a9a0ca532ac8
       Date/Time | 2023-11-29 16:18:21
   MPI Processes | 1
  OpenMP Threads | 256

 Reading settings XML file...
 Reading cross sections XML file...
 Reading materials XML file...
 Reading geometry XML file...
 Reading Li6 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Li6.h5
 Reading Li7 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Li7.h5
 Reading Be9 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Be9.h5
 Reading F19 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/F19.h5
 Reading Th232 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Th232.h5
 Reading U233 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/U233.h5
 Reading C0 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/C0.h5
 Reading B10 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/B10.h5
 Reading B11 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/B11.h5
 Reading Al27 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Al27.h5
 Reading Si28 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Si28.h5
 Reading Si29 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Si29.h5
 Reading Si30 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Si30.h5
 Reading P31 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/P31.h5
 Reading S32 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/S32.h5
 Reading S33 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/S33.h5
 Reading S34 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/S34.h5
 Reading S36 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/S36.h5
 Reading Ti46 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Ti46.h5
 Reading Ti47 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Ti47.h5
 Reading Ti48 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Ti48.h5
 Reading Ti49 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Ti49.h5
 Reading Ti50 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Ti50.h5
 Reading Cr50 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Cr50.h5
 Reading Cr52 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Cr52.h5
 Reading Cr53 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Cr53.h5
 Reading Cr54 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Cr54.h5
 Reading Mn55 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Mn55.h5
 Reading Fe54 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Fe54.h5
 Reading Fe56 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Fe56.h5
 Reading Fe57 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Fe57.h5
 Reading Fe58 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Fe58.h5
 Reading Co59 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Co59.h5
 Reading Ni58 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Ni58.h5
 Reading Ni60 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Ni60.h5
 Reading Ni61 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Ni61.h5
 Reading Ni62 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Ni62.h5
 Reading Ni64 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Ni64.h5
 Reading Cu63 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Cu63.h5
 Reading Cu65 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Cu65.h5
 Reading Nb93 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Nb93.h5
 Reading Mo92 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Mo92.h5
 Reading Mo94 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Mo94.h5
 Reading Mo95 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Mo95.h5
 Reading Mo96 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Mo96.h5
 Reading Mo97 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Mo97.h5
 Reading Mo98 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Mo98.h5
 Reading Mo100 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Mo100.h5
 Reading Hf174 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Hf174.h5
 Reading Hf176 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Hf176.h5
 Reading Hf177 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Hf177.h5
 Reading Hf178 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Hf178.h5
 Reading Hf179 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Hf179.h5
 Reading Hf180 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/Hf180.h5
 Reading W180 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/W180.h5
 Reading W182 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/W182.h5
 Reading W183 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/W183.h5
 Reading W184 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/W184.h5
 Reading W186 from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/W186.h5
 Reading c_Graphite from
 /home/cuiyong/OpenMC_Lib/Official_Data_Libraries/endfb71_hdf5/c_Graphite.h5
 Minimum neutron data temperature: 600 K
 Maximum neutron data temperature: 1200 K
 Reading tallies XML file...
 **ERROR: Could not find the nuclide U234 specified in tally 5 in any material**
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode -1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/cuiyong/mambaforge3/envs/saltproc-env/lib/python3.12/site-packages/saltproc/depcode.py", line 141, in run_depletion_step
    subprocess.run(
  File "/home/cuiyong/mambaforge3/envs/saltproc-env/lib/python3.12/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['python', '/home/cuiyong/mambaforge3/envs/saltproc-env/lib/python3.12/site-packages/saltproc/openmc_deplete.py', '--materials', '/home/cuiyong/Program_files/saltproc/examples/msbr/openmc_msbr_test/materials.xml', '--geometry', '/home/cuiyong/Program_files/saltproc/examples/msbr/openmc_msbr_test/geometry.xml', '--settings', '/home/cuiyong/Program_files/saltproc/examples/msbr/openmc_msbr_test/settings.xml', '--tallies', '/home/cuiyong/Program_files/saltproc/examples/msbr/openmc_msbr_test/tallies.xml', '--directory', '/home/cuiyong/Program_files/saltproc/examples/msbr/openmc_msbr_test']' returned non-zero exit status 255.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/cuiyong/Program_files/saltproc/saltproc/__main__.py", line 4, in <module>
    app.run()
  File "/home/cuiyong/mambaforge3/envs/saltproc-env/lib/python3.12/site-packages/saltproc/app.py", line 61, in run
    depcode.run_depletion_step(mpi_args, threads)
  File "/home/cuiyong/mambaforge3/envs/saltproc-env/lib/python3.12/site-packages/saltproc/openmc_depcode.py", line 401, in run_depletion_step
    super().run_depletion_step(mpi_args, args)
  File "/home/cuiyong/mambaforge3/envs/saltproc-env/lib/python3.12/site-packages/saltproc/depcode.py", line 150, in run_depletion_step
    print(error.output.decode("utf-8"))
          ^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'decode'

How can this problem be resolved? Thank you very much! 

In addition : Can I use the lower version (2.1.0) of serpent to perform MSR fuel cycle simulation via saltproc?

I'm looking forward to hearing from you soon. Best Regards!

**Comment by yardasol** [comment]
Hello @tsuiyong,
My apologies for the very late response. It seems that you are getting this error from OpenMC, not SaltProc. To confirm, can you run the base openmc model by executing `openmc` from the directory with your settings, material, geometry, and tally files?

**Reply by tsuiyong** [reply]
Yes, as you say, I can run the base openmc model of MSBR. The question I want to ask is which version of OpenMC is supported by the current version of  SaltProc? Is it the latest version or just 0.13.3? Thank you very much!

**Reply by yardasol** [reply]
Hi @tsuiyong, apologies again for the extremely long delay. It looks like the issue is that nuclide `U234`  is being tallied, but it is not found in any of your materials. Perhaps try using the latest version of OpenMC and try running the simulation again.

**Reply by tsuiyong** [reply]
Dear @yardasol,

I deeply appreciate your generous sharing of the excellent code. I have successfully run the code with the MSBR example using the latest version of OpenMC 0.14.0. However, I have encountered an obstacle in processing the 121-step data using the provided scripts in ~/saltproc/scripts, such as burnup_plotter.ipynb, keff_plotter.ipynb, nuclides_mass.ipynb, and others, due to the specific path /home/andrei2/Desktop/git/meitner-task-2/saltproc_results/feed_leu_19.79/db_saltproc.h5.

Given my time constraints, I kindly request your assistance in sharing the script for generating the db_saltproc.h5 file. Your support in this matter would be immensely valuable. 

Thank you for your understanding and support, and your generous help will be acknowledged in my future work.

Warm regards!

**Reply by yardasol** [reply]
Hi @tsuiyong,

Unfortunately, the many of  the scripts in the `scripts` directory are outdated and are unsupported. There is an open issue (#91) to address this, so if you figure them out and find a solution feel free to make a pull request with your changes!

`db_saltproc.h5` is just the results file that SaltProc creates. Our [online docpages](https://arfc.github.io/saltproc/master/io_formats/databasefile.html) should be able to help you figure out the structure of this file to write your own analysis scripts. You can also look int the test suite to see examples of how we interface with the `.h5` file. [Here](https://github.com/arfc/saltproc/blob/master/tests/integration_tests/run_constant_reprocessing_openmc/test.py) is one example.

**Reply by tsuiyong** [reply]
Hi @yardasol, thank you for your generous help, I will address this issue according to your advice!
